### PR TITLE
Wear and Tear: Light Bulbs and Switches

### PR DESCRIPTION
**Pull request by pgmzeta**
[FEATURE][INPUT WANTED][STATION SYSTEMS][BALANCE]
## About the PR 

* Light bulbs have a 1% probability of "burning out" when activated, doing the following:
    * The light bulb changes to the burned out sprite, mostly filling the fixture
    * The light bulb does not emit light
    * Burnt-out light bulbs have a 6.25% probability of exploding when activated
* Light switches slowly "wear out" over time:
    * After ten uses, can shock the user
    * After fifty uses, will also spark at the user's location
    * Will only shock/spark when used twice in fifteen seconds.
    * Using a screwdriver on a light switch resets the usage to zero

## Light bulb bursting effect 
https://github.com/goonstation/goonstation/assets/91498627/90fa2cf8-59ba-408d-8b6b-30704a482ea1
Last light switch toggle burns out the left floor bulb

## Why's this needed? 
Give station lighting infrastructure a rate of "decay" incurred by use, enabling players to weaponize and react to the station "running down" over the course of a shift. Variance in light burn out allows for the station to wear down in minor ways over the course of a round. Small increase of impact to things that mess with APC power e.g. arcfiends / power sinks / ID access / cutting wires, and some random events that toggle/effect lights i.e. Vamp TEG, ion storm.

## TODO
- [x] Add vfx/sfx for bulb burning out

## Test Plan
- [x] Light switch shock
  - [x] Spam on/off shocks
  - [x] Waiting for the cooldown does not shock
  - [x] Shock still happens if cooldown is re-triggered
  - [x] Screwdriver with light switch on shocks user
  - [x] Insuls protect from screwdriver shock
  - [x] Screwdriver with light switch off does not shock user
- [x] Light bulb burn out
  - [x] Bulb item and icon sprite stay in sync when burned out
  - [x] Burned out lights do not emit light
  - [x] Burned out bulbs are replaceable with a portable lamp manufacturer

## Changelog 

```changelog
(u)glowbold
(*)NanoTrasen has secured further discounts on station electrical equipment - details in minor notes.
(+)Lights may burn out when turned on. Burned out lights may burst if left unfixed.
(+)Over-use of light switches may cause a shock to users and/or their property. Re-secure the electrical contacts on light switches with a screwdriver.
```

### CODE REVIEW

**Comment by pali6** [comment]
Going full circle with lightbulbs, heh

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -798,20 +801,41 @@ ADMIN_INTERACT_PROCS(/obj/machinery/light, proc/broken, proc/admin_toggle, proc/
...
+	logTheThing(LOG_STATION, null, "Light '[name]' broke itself (breakprob: [current_lamp.breakprob]) at ([log_loc(src)])")
 
+/obj/machinery/light/proc/do_burn_out()
+	current_lamp.breakprob = WORN_LIGHT_BREAKPROB
+	current_lamp.light_status = LIGHT_BURNED
+	current_lamp.update()
+	icon_state = "[base_state]-burned"
+	on = 0
+	light.disable()
+	logTheThing(LOG_STATION, null, "Light '[name]' burned out (burnprob: [current_lamp.burnprob]) at ([log_loc(src)])")
```
Comment (Tarmunora) [inline code/modules/power/lighting.dm:838]: could probably use some light sfx or something to make it clearer that A Thing Happened

**Review by Tarmunora — APPROVED** [review]

File: code/modules/power/lighting.dm (hunk 1 above)
Comment (pgmzeta) [reply code/modules/power/lighting.dm]: Added a brightness flash and sfx for popping, video in desc.

**Comment by Tarmunora** [comment]
lgtm

**Pull request merged**